src/consensus/pbft/libbyz/client_proxy.h

```diff
@@ -230,6 +236,15 @@ bool ClientProxy<T, C>::send_request(
   req->authenticate(len, is_read_only);
 
   auto req_clone = req->clone();
+  try
+  {
+    req_clone->create_context(verify_command);
+  }
+  catch (const std::exception& e)
+  {
+    LOG_FAIL_FMT("Failed to parse arguments, e.what:", e.what());
```

Comment:
We need to do a pass on all logging to set levels correctly, because logging this kind of error as FAIL potentially amounts to revealing a request to the host. I'm not singling out this particular call, we have other instances, just that we should define a good pattern for those in general.

I suggest we essentially use the following convention: all INFO, ERROR and FAIL lines should be scrubbed of potentially confidential info. They can either be supplemented with a separate DEBUG call with details, or we come up with a way to pass confidential and non-confidential args separately, and when the level is higher than debug, confidential args are replaced with *** or some other placeholder.

I can't think of a good way to automatically check for this/prevent accidental logging of confidential data, if someone has a good idea, I'm very curious to hear it!

Reply:
In addition to that, the log level needs to be attested and checked on joining nodes.